### ISSUE TITLE
lx.executeParentOriginXHR is not honoring the method input argument

### ISSUE THREAD

**Issue by pauloramires** [leanix/leanix-reporting#16]
When using the [executeParentOriginXHR](https://leanix.github.io/leanix-reporting/classes/lxr.lxcustomreportlib.html#executeparentoriginxhr) to make a request inside the custom-report iframe, the *method* input argument is not honored. The image attached below depicts a network analysis screenshot for the following requests: **lx.executeParentOriginXHR('GET', *'the call url'*)** and **lx.executeParentOriginXHR('POST', *'the call url'*)**

![screenshot_methods](https://user-images.githubusercontent.com/1428629/60969456-f0abf800-a2f5-11e9-92fa-6c9fc1e7fc16.png)

Version tested: [leanix-reporting v0.3.3](https://github.com/leanix/leanix-reporting/releases/tag/0.3.3)

**Comment by rene-leanix** [comment]
You are right. @pauloramires! I've also stumbled across this today and adjusted the documentation for this method accordingly. We are even considering removing this method altogether due to security implications. Are you currently using it in any of your reports?

**Comment by pauloramires** [comment]
@rene-leanix hi! I am not using it in any report although I think that it could be useful to interface webhooks api in order to provide some real time interactivity features in the custom reports

**Comment by CroniD** [comment]
Hi @rene-leanix ,
the documentation for "executeParentOriginXHR" doesn't mention that only GET is supported as stated by you and the release notes.

Furthermore, incowia wants to use this to read and save data to metrics inside a custom report. Its an open topic for us since 2 years. :(

And, lets not forget, it is actually pretty useful, when reading data from another system outside of leanix, but still want to know if the call is allowed to do so (verifying its an leanix user), so the other system just have to check the bearer token and must not act on its own by using an API token internally and asking leanix if the user exists for that workspace.

**Comment by rene-leanix** [comment]
Thanks for your feedback, @CroniD! I assume you'd require a PUT or POST to achieve this? We'll need to discuss the security implications and your requirements internally.

You're right about the documentation. There was a spelling mistake in the JSDoc used to generate the documentation. This will be updated today.

**Comment by CroniD** [comment]
Hi @rene-leanix ,

you're right, but I would suggest to limit the call to just GET (maybe including other read methods), when calling non-leanix systems, but open to all methods that are needed to call known leanix APIs. That would be a compromis that'll work for us. :)
